File: backend/GarmentsERP.API/Services/BusinessSettingService.cs
using Microsoft.EntityFrameworkCore;
using GarmentsERP.API.Data;
using GarmentsERP.API.DTOs.Settings;
using GarmentsERP.API.Interfaces;
using GarmentsERP.API.Models.Settings;

namespace GarmentsERP.API.Services
{
    public class BusinessSettingService : IBusinessSettingService
    {
        private readonly ApplicationDbContext _context;

        public BusinessSettingService(ApplicationDbContext context)
        {
            _context = context;
        }

        public async Task<IEnumerable<BusinessSettingDto>> GetAllBusinessSettingsAsync()
        {
            return await _context.BusinessSettings
                .Select(bs => new BusinessSettingDto
                {
                    Id = bs.Id,
                    SettingKey = bs.SettingKey,
                    SettingValue = bs.SettingValue,
                    Description = bs.Description,
                    Category = bs.Category,
                    CategoryName = bs.Category.ToString(),
                    IsSystem = bs.IsSystem,
                    CreatedAt = bs.CreatedAt,
                    UpdatedAt = bs.UpdatedAt
                })
                .OrderBy(bs => bs.Category)
                .ThenBy(bs => bs.SettingKey)
                .ToListAsync();
        }

        public async Task<BusinessSettingDto?> GetBusinessSettingByIdAsync(Guid id)
        {
            return await _context.BusinessSettings
                .Where(bs => bs.Id == id)
                .Select(bs => new BusinessSettingDto
                {
                    Id = bs.Id,
                    SettingKey = bs.SettingKey,
                    SettingValue = bs.SettingValue,
                    Description = bs.Description,
                    Category = bs.Category,
                    CategoryName = bs.Category.ToString(),
                    IsSystem = bs.IsSystem,
                    CreatedAt = bs.CreatedAt,
                    UpdatedAt = bs.UpdatedAt
                })
                .FirstOrDefaultAsync();
        }

        public async Task<BusinessSettingDto?> GetBusinessSettingByKeyAsync(string key)
        {
            return await _context.BusinessSettings
                .Where(bs => bs.SettingKey == key)
                .Select(bs => new BusinessSettingDto
                {
                    Id = bs.Id,
                    SettingKey = bs.SettingKey,
                    SettingValue = bs.SettingValue,
                    Description = bs.Description,
                    Category = bs.Category,
                    CategoryName = bs.Category.ToString(),
                    IsSystem = bs.IsSystem,
                    CreatedAt = bs.CreatedAt,
                    UpdatedAt = bs.UpdatedAt
                })
                .FirstOrDefaultAsync();
        }

        public async Task<BusinessSettingDto> CreateBusinessSettingAsync(CreateBusinessSettingDto createDto)
        {
            // Check if key already exists
            var existingSetting = await _context.BusinessSettings
                .FirstOrDefaultAsync(bs => bs.SettingKey == createDto.SettingKey);
            
            if (existingSetting != null)
                throw new InvalidOperationException($"Setting with key '{createDto.SettingKey}' already exists");

            var businessSetting = new BusinessSetting
            {
                SettingKey = createDto.SettingKey,
                SettingValue = createDto.SettingValue,
                Description = createDto.Description,
                Category = createDto.Category,
                IsSystem = createDto.IsSystem
            };

            _context.BusinessSettings.Add(businessSetting);
            await _context.SaveChangesAsync();

            return await GetBusinessSettingByIdAsync(businessSetting.Id) ?? 
                   throw new InvalidOperationException("Failed to retrieve created business setting");
        }

        public async Task<BusinessSettingDto?> UpdateBusinessSettingAsync(Guid id, UpdateBusinessSettingDto updateDto)
        {
            var businessSetting = await _context.BusinessSettings.FindAsync(id);
            if (businessSetting == null)
                return null;

            // Prevent updating system settings
            if (businessSetting.IsSystem)
                throw new InvalidOperationException("Cannot update system settings");

            businessSetting.SettingValue = updateDto.SettingValue;
            businessSetting.Description = updateDto.Description;
            businessSetting.UpdatedAt = DateTime.UtcNow;

            await _context.SaveChangesAsync();
            return await GetBusinessSettingByIdAsync(id);
        }

        public async Task<bool> DeleteBusinessSettingAsync(Guid id)
        {
            var businessSetting = await _context.BusinessSettings.FindAsync(id);
            if (businessSetting == null)
                return false;

            // Prevent deleting system settings
            if (businessSetting.IsSystem)
                throw new InvalidOperationException("Cannot delete system settings");

            _context.BusinessSettings.Remove(businessSetting);
            await _context.SaveChangesAsync();
            return true;
        }

        public async Task<IEnumerable<BusinessSettingDto>> GetBusinessSettingsByCategoryAsync(SettingCategory category)
        {
            return await _context.BusinessSettings
                .Where(bs => bs.Category == category)
                .Select(bs => new BusinessSettingDto
                {
                    Id = bs.Id,
                    SettingKey = bs.SettingKey,
                    SettingValue = bs.SettingValue,
                    Description = bs.Description,
                    Category = bs.Category,
                    CategoryName = bs.Category.ToString(),
                    IsSystem = bs.IsSystem,
                    CreatedAt = bs.CreatedAt,
                    UpdatedAt = bs.UpdatedAt
                })
                .OrderBy(bs => bs.SettingKey)
                .ToListAsync();
        }

        public async Task<string?> GetSettingValueAsync(string key)
        {
            return await _context.BusinessSettings
                .Where(bs => bs.SettingKey == key)
                .Select(bs => bs.SettingValue)
                .FirstOrDefaultAsync();
        }

        public async Task<bool> UpdateSettingValueAsync(string key, string value)
        {
            var businessSetting = await _context.BusinessSettings
                .FirstOrDefaultAsync(bs => bs.SettingKey == key);
            
            if (businessSetting == null)
                return false;

            // Prevent updating system settings
            if (businessSetting.IsSystem)
                throw new InvalidOperationException("Cannot update system settings");

            businessSetting.SettingValue = value;
            businessSetting.UpdatedAt = DateTime.UtcNow;

            await _context.SaveChangesAsync();
            return true;
        }
    }
}

File: frontend/src/services/businessSettingService.ts
import { apiService } from './apiService';

export interface BusinessSettingDto {
  id: string;
  settingKey: string;
  value: string;
  category: string;
  description?: string;
  dataType: string;
  isSystem: boolean;
  isActive: boolean;
  validationRules?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateBusinessSettingDto {
  settingKey: string;
  value: string;
  category: string;
  description?: string;
  dataType: string;
  isSystem: boolean;
  isActive: boolean;
  validationRules?: string;
}

export interface UpdateBusinessSettingDto {
  settingKey: string;
  value: string;
  category: string;
  description?: string;
  dataType: string;
  isSystem: boolean;
  isActive: boolean;
  validationRules?: string;
}

export const businessSettingService = {
  async getAll(): Promise<BusinessSettingDto[]> {
    return apiService.get<BusinessSettingDto[]>('/api/BusinessSetting');
  },

  async getById(id: string): Promise<BusinessSettingDto> {
    return apiService.get<BusinessSettingDto>(`/api/BusinessSetting/${id}`);
  },

  async getByKey(key: string): Promise<BusinessSettingDto> {
    return apiService.get<BusinessSettingDto>(`/api/BusinessSetting/by-key/${key}`);
  },

  async getByCategory(category: string): Promise<BusinessSettingDto[]> {
    return apiService.get<BusinessSettingDto[]>(`/api/BusinessSetting/by-category/${category}`);
  },

  async getSettingValue(key: string): Promise<string> {
    return apiService.get<string>(`/api/BusinessSetting/value/${key}`);
  },

  async create(data: CreateBusinessSettingDto): Promise<BusinessSettingDto> {
    return apiService.post<BusinessSettingDto>('/api/BusinessSetting', data);
  },

  async update(id: string, data: UpdateBusinessSettingDto): Promise<BusinessSettingDto> {
    return apiService.put<BusinessSettingDto>(`/api/BusinessSetting/${id}`, data);
  },

  async updateSettingValue(key: string, value: string): Promise<void> {
    return apiService.put<void>(`/api/BusinessSetting/value/${key}`, value, {
      headers: {
        'Content-Type': 'application/json'
      }
    });
  },

  async delete(id: string): Promise<void> {
    return apiService.delete<void>(`/api/BusinessSetting/${id}`);
  }
};
